packages/fxa-auth-server/lib/db/index.ts

```diff
@@ -0,0 +1,1126 @@
+      log.trace('DB.securityEvent', {
+        securityEvent: event,
+      });
+      await SecurityEvent.create({
+        ...event,
+        ipHmacKey: config.securityHistory.ipHmacKey,
+      });
+    }
+
+    async securityEvents(params: { uid: string; ipAddr: string }) {
+      log.trace('DB.securityEvents', {
+        params: params,
+      });
+      const { uid, ipAddr } = params;
+      return SecurityEvent.findByUidAndIP(
+        uid,
+        ipAddr,
+        config.securityHistory.ipHmacKey
+      );
+    }
+
+    async securityEventsByUid(params: { uid: string }) {
+      log.trace('DB.securityEventsByUid', {
+        params: params,
+      });
+      const { uid } = params;
+      return SecurityEvent.findByUid(uid);
+    }
+
+    async createUnblockCode(uid: string): Promise<any> {
+      if (!UnblockCode) {
+        throw new Error('Unblock has not been configured');
+      }
+      log.trace('DB.createUnblockCode', { uid });
+      const code = await UnblockCode();
+      try {
+        await Account.createUnblockCode(uid, code);
+        return code;
+      } catch (err) {
+        // duplicates should be super rare, but it's feasible that a
+        // uid already has an existing unblockCode. Just try again.
+        if (isRecordAlreadyExistsError(err)) {
+          log.error('DB.createUnblockCode.duplicate', {
+            err: err,
+            uid: uid,
+          });
+          return this.createUnblockCode(uid);
+        }
+        throw err;
+      }
+    }
+
+    async consumeUnblockCode(uid: string, code: string) {
+      log.trace('DB.consumeUnblockCode', { uid });
+      try {
+        return await Account.consumeUnblockCode(uid, code);
+      } catch (err) {
+        if (isNotFoundError(err)) {
+          throw error.invalidUnblockCode();
+        }
+        throw err;
+      }
+    }
+
+    async createEmailBounce(
+      bounceData: Parameters<typeof EmailBounce.create>[0]
+    ) {
+      log.trace('DB.createEmailBounce', {
+        bounceData: bounceData,
+      });
+      await EmailBounce.create(bounceData);
+    }
+
+    async createEmail(uid: string, emailData: any) {
+      log.trace('DB.createEmail', {
+        email: emailData.email,
+        uid,
+      });
+
+      try {
+        await Account.createEmail({ uid, ...emailData });
+      } catch (err) {
+        if (isEmailAlreadyExistsError(err)) {
+          throw error.emailExists();
+        }
+        throw err;
+      }
+    }
+
+    async deleteEmail(uid: string, email: string) {
+      log.trace('DB.deleteEmail', { uid });
+
+      try {
+        return await Account.deleteEmail(uid, email);
+      } catch (err) {
+        if (isEmailDeletePrimaryError(err)) {
+          throw error.cannotDeletePrimaryEmail();
+        }
+        throw err;
+      }
+    }
+
+    async createSigninCode(uid: string, flowId: string): Promise<any> {
+      log.trace('DB.createSigninCode');
+
+      const code = await random.hex(config.signinCodeSize);
```

Comment:
I find the type here confusing. I think it will be a string, at least that's what Account.createSigninCode expects...